### ISSUE TITLE
Question about logexpsum

### ISSUE THREAD

**Issue by renzilin** [tryambak2019/SGAN#1]
Dear author,
I have a question about the logexpsum in your code.
![image](https://user-images.githubusercontent.com/17135904/86866860-c2473100-c09f-11ea-88b1-f05e90dd6f7d.png)

In the picuture, your code is trying to do the sum(exp(x)) / (sum(exp(x)) + 1). But in the nn.logsumexp, there is a log operation. Is that the same when dealing this loss? 

Could you please explain it?
Thank you so much with your help.

Best,
Zilin

**Comment by tryambak2019** [comment]
Hi Zilin,

I use logexpsum only as a temporary variable to calculate variable adv. The adv variable is basically modified sigmoid output (distribute values between (0,1) and which not necessarily add to 1) and used for 2-class logistic regression with BCELoss.

The variable aux calculates softmax output (distribute values within (0,1) to sum upto 1) and is used for multiclass regression with CrossEntrpyLoss.

nn.logsumexp does not necessarily maps the values to interval (0,1) and hence cannot be used directly to evaluate logistic regressions (with BCELoss and CrossEntropyLoss).

Hope it helps !

**Comment by renzilin** [comment]
Thank you so much. It helps a lot!
